modules/ListView.js

```diff
@@ -367,48 +417,110 @@
-    __createChildViews: function() {
-      var newChildViews = [];
-      _.each(this.modelsToRender(), function(model) {
+    __createItemViews: function() {
+      var newItemViews = [];
+      _.each(this.modelsToRender(), function(model, indexOfModel) {
         var childView = this.getChildViewFromModel(model);
         if (!childView) {
-          newChildViews.push(this.__createChildView(model));
+          newItemViews.push({ view: this.__createItemView(model, true), indexOfModel: indexOfModel });
         }
       }, this);
-      return newChildViews;
+      this.__updateOrderedModelIdList();
+      return newItemViews;
     },
 
     /**
-     * Removes a child view's that have models that are no longer tracked by modelsToRender
-     * @method __removeStaleChildViews
+     * Creates an item view and stores a reference to it
+     * @method __createItemView
      * @private
+     * @param model {Backbone Model} the model to create the view from
+     * @param [noUpdateToIdList=false] if true, the internal order of model ids are not updated
+     * @return {Backbone View} the new item view
      */
-    __removeStaleChildViews: function() {
-      var removedChildViews = [];
+    __createItemView: function(model, noUpdateToIdList) {
+      var itemView,
+        ItemViewClass = this.childView;
+      if (!_.isFunction(this.childView.extend)) {
+        ItemViewClass = this.childView(model);
+      }
+      itemView = new ItemViewClass(this.__generateChildArgs(model));
+      this.registerTrackedView(itemView, { shared: false });
+      this.__modelToViewMap[model[this.__modelId]] = itemView.cid;
+      if (!noUpdateToIdList) {
+        this.__updateOrderedModelIdList();
+      }
+      this.trigger('child-view-added', {model: model, view: itemView});
+      return itemView;
+    },
+
+    /**
+     * Gets all item views that have models that are no longer tracked by modelsToRender
+     * @method __getStaleItemViews
+     * @return {Array} An array of information about stale items. Each object has a 'view' and 'modelId' field
+     * @private
+     */
+    __getStaleItemViews: function() {
+      var staleItemViews = [];
       var modelsWithViews = _.clone(this.__modelToViewMap);
       _.each(this.modelsToRender(), function(model) {
-        var childView = this.getChildViewFromModel(model);
-        if (childView) {
+        var itemView = this.getChildViewFromModel(model);
+        if (itemView) {
           delete modelsWithViews[model[this.__modelId]];
         }
       }, this);
       _.each(modelsWithViews, function(viewId, modelId) {
-        var childView = this.getChildView(viewId);
-        if (childView) {
-          removedChildViews.push(childView);
-          _removeChildView.call(this, childView, modelId);
+        var itemView = this.getChildView(viewId);
+        if (itemView) {
+          staleItemViews.push({ view: itemView, modelId: modelId });
         } else {
           delete this.__modelToViewMap[modelId];
```

Comment:
should i remove it or update the docs? I just thought I'd clean up if somehow the state gets off